<launch>
  <arg name="voxel_size" default="0.2"/>
  <arg name="voxels_per_side" default="16" />
  <arg name="truncation_distance" default="0.6" />
  <arg name="ray_length" default="5" />

  <arg name="UAV_NAME" default="$(optenv UAV_NAME uav1)" />
  <arg name="world_frame" default="$(arg UAV_NAME)/world_origin" />

  <group ns="$(arg UAV_NAME)">
    <node pkg="motion_planning" type="GAEP" name="planner_node" output="screen">
      <rosparam file="$(find rrt_construction)/config/GainConfig.yaml" />
      <rosparam file="$(find motion_planning)/config/GAEPlanner.yaml" />
      <param name="uav_namespace" value="$(arg UAV_NAME)" />
      <!-- Voxblox Planner -->
      <remap from="~esdf_map_in" to="voxblox_node/esdf_map_out" />
      <remap from="~tsdf_map_in" to="voxblox_node/tsdf_map_out" />
      <param name="use_tf_transforms" value="false" />
      <param name="tsdf_voxel_size" value="$(arg voxel_size)" />
      <param name="tsdf_voxels_per_side" value="$(arg voxels_per_side)" />
      <param name="update_mesh_every_n_sec" value="0.0" />
      <param name="publish_traversable" value="false" />
      <param name="publish_pointclouds" value="false" />
      <param name="max_ray_length" value="$(arg ray_length)" />
      <!-- Publishers -->
      <remap from="~visualization_marker_out" to="visualization_marker_out" />
      <remap from="~initial_reference_out" to="control_manager/reference" />
      <remap from="~simulation_ready" to="simulation_ready" />
      <remap from="~reference_out" to="reference_out" />
      <remap from="~tree_node_out" to="tree_node" />
      <remap from="~frustum_out" to="frustum_out" />
      <remap from="~unknown_voxels_out" to="unknown_voxels_out" />
      <!-- Subscribers -->
      <!--remap from="~constraints_in" to="control_manager/current_constraints" /-->
      <remap from="~uav_state_in" to="estimation_manager/uav_state" />
      <remap from="~tracker_cmd_in" to="control_manager/tracker_cmd" />
      <remap from="~control_manager_diag_in" to="control_manager/diagnostics" />
      <!-- Services clients-->
      <remap from="~trajectory_generation_out" to="trajectory_generation/get_path" />
      <remap from="~trajectory_reference_out" to="control_manager/trajectory_reference" />
      <remap from="~best_node_out" to="cache_nodes/best_node" />
      <remap from="~gp_query_out" to="cache_nodes/gp_query" />
      <!--remap from="~path_out" to="trajectory_generation/path" /-->
      <!-- Services servers-->
      <remap from="~start_in" to="~start" />
      <remap from="~stop_in" to="~stop" />
      <!--remap from="~reevaluate_in" to="~reevaluate" /-->

    </node>

  </group>

</launch>
